authenticity.cpp

```diff
@@ -189,19 +185,19 @@ std::string Authenticity::Assay
         }
 
     // Validate all data files.
-    fs::path original_path(fs::current_path());
-    if(0 != chdir(data_path.string().c_str()))
-        {
-        oss
-            << "Unable to change directory to '"
-            << data_path
-            << "'. Try reinstalling."
-            ;
-        return oss.str();
-        }
     try
         {
-        system_command("md5sum --check --status " + std::string(md5sum_file()));
+        auto const md5sums_path = data_path / md5sum_file();
+        auto const sums = md5_read_checksum_file(md5sums_path);
+        for(auto const& s : sums)
+            {
+            auto const file_path = data_path / s.filename;
+            auto const md5 = md5_calculate_checksum(file_path);
+            if(md5 != s.md5sum)
+                {
+                    throw std::runtime_error("Error validating MD5 sum");
```

Comment:
I'd probably avoid mentioning "MD5 sum" here (does the current version say anything about MD5 if the check fails?), but I would include the file name in the error message. E.g. something like

    throw std::runtime_error("Integrity check failed for '" + s.filename + "'");